### ISSUE TITLE
no autocomplete in macro arguments

### ISSUE THREAD

**Issue by Milo123459** [rust-lang/rust-analyzer#15851]
i was using a macro, and had no autocomplete for a local `interaction` variable. outside of the macro arguments, i had autocomplete for it.

**rust-analyzer version**: 2023-11-06

**rustc version**: 1.73.0

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)

**Comment by flodiebold** [comment]
This depends heavily on the actual macro and the code inside it. Can you give a reproduction?

**Comment by Milo123459** [comment]
it is the `info` macro from `tracing` v0.1.40
```rs
   interaction.user.id.0 as i64; // autocomplete
                        info!("Command `{}` was run by {}", interaction.data.name, interaction.user.id.0); // no autocomplete on interaction
```

**Comment by Veykril** [comment]
Iirc, tracing uses the macro args multiple times where the first usage gets plugged into a place where we can't figure out anything semantically and so we fail completions then (as well as highlighting etc.)

**Comment by flodiebold** [comment]
Maybe we could at least continue trying other places in those situations.

**Comment by Milo123459** [comment]
is there anyway i can help out with this?

**Comment by Veykril** [comment]
> Maybe we could at least continue trying other places in those situations.

Yes, we should in general -> https://github.com/rust-lang/rust-analyzer/issues/11058. "Just" checking for this case is anything but easy though, even heuristically I don't see how we can figure out these kinds of cases easily.

As context, the problematic macro is [`event!`](https://docs.rs/tracing/latest/src/tracing/macros.rs.html#585-1023), note how `fields` is being reused a bunch, and the first usage is passed to [`callsite2!`](https://docs.rs/tracing/latest/src/tracing/macros.rs.html#2768) which passes it to [`fieldset!`](https://docs.rs/tracing/latest/src/tracing/macros.rs.html#2991) stringifying all the arguments.

> is there anyway i can help out with this?

I'm personally unsure how we can proceed here currently unfortunately.

**Comment by ChayimFriedman2** [comment]
`tracing` macros are completion-friendly now.
